src/alarm/alarm_manager.cc

```diff
@@ -54,7 +54,8 @@ int AlarmManager::ScheduleAlarm(const std::string& app_id,
   }
 
   int alarm_id = 0;
-  if (alarm->type() == AlarmInfo::ABSOLUTE) {
```

Comment:
Fair enough. In this case, how about using a `switch` instead, as you're dealing with an enum?